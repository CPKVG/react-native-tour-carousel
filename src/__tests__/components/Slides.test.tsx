import React from 'react';
import { render } from '@testing-library/react-native';
import Slides from '../../components/Slides';

describe('<Slides />', () => {
  const props = {
    data: { title: '', subtitle: '', illustration: '' },
    itemHorizontalMargin: 10,
    slideConfig: {
      slideInnerContainer: {},
      slideImageContainer: {},
      slideTextContainer: {},
      slideTitleStyle: {},
      slideSubtitleStyle: {},
      maxNumberOfLines: 1,
    },
  };
  it('should render without crashing when provided with minimal valid props', () => {
    const { root } = render(<Slides {...props} />);
    expect(root).toBeTruthy();
  });
  it('renders for a Slides correctly', () => {
    const Renderer = render(<Slides {...props} />);
    expect(Renderer.toJSON()).toMatchSnapshot();
  });
});
