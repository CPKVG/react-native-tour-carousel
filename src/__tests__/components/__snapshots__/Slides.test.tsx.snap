// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slides /> renders for a Slides correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={{}}
  >
    <Image
      source={
        {
          "uri": "",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "resizeMode": "contain",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {},
        {
          "height": 400.2,
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={{}}
    />
    <Text
      numberOfLines={1}
      style={{}}
    />
  </View>
</View>
`;
